manager_dashboard/manager_dashboard/js/uploadProjects.js

```diff
@@ -164,20 +184,50 @@ function check_imagery_url() {
 }
 
 
-function upload_to_firebase() {
+function uploadToFirebase() {
     switch (currentUid) {
         case null:
             alert("You are not logged in.");
         default:
-            // get form data
             // TODO: add checks if all input values are valid, e.g. image available
-            if (check_imagery_url() === false) {
+            if (checkImageryUrl() === false) {
                 console.log("could not create project due to imagery url.")
             }
             else {
+                 // get form data
                 mapswipe_import = getFormInput()
                 // upload projectDraft to firebase once image has been uploaded
-                upload_project_image(mapswipe_import)
+                // if inputGeometries is a TMId, check if it is a valid project, only upload if it is
+                var modal = document.getElementById("uploadModal");
+                modal.style.display = "block";
+                var modalValidating = document.getElementById("modalValidating");
+                modalValidating.style.display = "block";
+                switch(mapswipe_import.inputType){
+                    case "link":
+                        uploadProjectImage(mapswipe_import)
+                        break;
+                    case "aoi_file":
+                    console.log("here i am rock you liek a hurricane")
+                        answer = countObjectsInFilter(mapswipe_import.geometry, mapswipe_import.filter)
+                        if (answer.Error != null){
+                            alert(`Invalid TMId: ${answer.Error}`)
+                        }
+                        else {
+                            uploadProjectImage(mapswipe_import)
+                        }
+                        break;
+                    case "TMId":
```

Comment:
what is the expected behaviour here. When I submit a TM project ID should we check also if there are any buildings. as you do above?

I would assume that we should do 2 checks here.

Reply:
we already do that